_posts/2023-08-15-securing-wildfly-apps-saml-openshift.adoc

```diff
@@ -0,0 +1,382 @@
+this servlet using SAML.
+
+You'll need to fork and then clone this Git repository: https://github.com/wildfly-security-incubator/elytron-examples.git
+
+We will be using the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-saml[simple-webapp-saml] directory in this repo.
+
+== Log Into the OpenShift Cluster
+
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]:
+
+[source]
+----
+oc login -u myUserName
+----
+
+Alternatively, you can log in using an API token:
+
+[source]
+----
+oc login --token=myToken --server=myServerUrl
+----
+
+You can request the token via the *Copy Login Command* link in the OpenShift web console.
+
+If you don't already have a project created, you can create one using:
+
+[source]
+----
+oc new-project myProjectName
+----
+
+== Start Keycloak
+
+We will be using Keycloak for our SAML identity provider.
+
+To start a Keycloak server in your project on OpenShift, use the following command:
+
+[source]
+----
+oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml \
+    -p KEYCLOAK_ADMIN=admin \                 // <1>
+    -p KEYCLOAK_ADMIN_PASSWORD=admin \        // <2>
+    -p NAMESPACE=myNameSpace \                // <3>
+| oc create -f -
+----
+<1> Replace `admin` with the user name you would like to use when accessing the Keycloak Administration Console.
+<2> Replace `admin` with the password you would like to use when accessing the Keycloak Administration Console.
+<3> Replace `myNameSpace` with your project name.
+
+After running the above command, you should see the following output:
+
+[source]
+----
+service/keycloak created
+route.route.openshift.io/keycloak created
+deploymentconfig.apps.openshift.io/keycloak created.
+----
+
+It will take a few minutes for OpenShift to provision the Keycloak pod and its related resources.
+
+You can use the OpenShift CLI or the OpenShift web console, depending on your preference,
+to check if your Keycloak server has been provisioned.
+
+=== OpenShift CLI
+
+To make sure your Keycloak server has been provisioned using the OpenShift CLI, run:
+
+[source]
+----
+oc get pods
+----
+
+After a little while, check for a message similar to the following message that indicates the pod is ready:
+
+[source]
+----
+NAME                READY     STATUS      RESTARTS   AGE
+keycloak-1-deploy   0/1       Completed   0          1h
+keycloak-1-l9kdx    1/1       Running     0          1h
+----
+
+Once the Keycloak server has been provisioned, use the following command to find the URL your Keycloak instance's
```

Comment:
@fjuma Total minor, should this be `find the URL of your Keycloak instance's Admin Console:` ?

Reply:
Good catch, fixed, thanks!